### PR TITLE
Feat: Join a group service

### LINKED ISSUE #206: Join a group service
## Feature Title

Join a group service

## Acceptance Criteria

- [x] Join a group service

## Technical Details

- API Changes:
- Database Changes:
- Dependencies:

## Additional Context

None

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Added a `join to group by hash` service 

## Review points

I've also added the validator of the MD5.

### CODE REVIEW

File: backend/main/src/Services/InvitationHashService/InvitationHashService.ts [hunk 1]
```diff
@@ -127,7 +118,49 @@ export class InvitationHashService implements IGenerateAnInvitationHashService {
...
+		]);
+
+		if (!userData) {
+			return Err(new UserNotFound("Requesting user is not found."));
+		}
+
+		if (!invitationLinkHash) {
+			return Err(new HashNotFound("Invalid hash is requested."));
+		}
+
```
Comment (taniguchiiqqq) [inline backend/main/src/Services/InvitationHashService/InvitationHashService.ts:152]: I think we need check if the invitationLinkHash.linkExpiryTime has already passed. and if the link expires, the specific error should be returned.

File: backend/main/src/Services/InvitationHashService/InvitationHashService.ts [hunk 2]
```diff
@@ -127,7 +118,49 @@ export class InvitationHashService implements IGenerateAnInvitationHashService {
 			expiryDatetime,
 		});
 	}
+
+	/**
+	 * This method join a user to the group using an invitation hash.
+	 * This method checks the existence of the invitation hash and the user.
+	 * If the invitation hash and the user exists, this method adds the user to the group.
+	 * And return the groupId of the group.
```
Comment (taniguchiiqqq) [inline backend/main/src/Services/InvitationHashService/InvitationHashService.ts:125]: The comment is also going to be updated to mention that this function checks If the link's expiry datetime is over.

File: backend/main/test/Services/InvitationHashService.test.ts
Comment (taniguchiiqqq) [inline backend/main/test/Services/InvitationHashService.test.ts:1]: I think it's better to add a test case where the hash exist and expiry datetime exceed.

**Review by taniguchiiqqq — CHANGES_REQUESTED** [review]
Thank you! I commented about the case where the expiry datetime already has passed.

**Comment by hitohata** [comment]
@taniguchiiqqq 
I'm sorry. I completely missed checking the link expiry time.
Could you check this PR again?

**Review by taniguchiiqqq — APPROVED** [review]
Thank you for fixing it! LGMT

**Pull request merged**